### DISCUSSION TITLE
[INFO] [auth] [2022-06-23T06:52:51.250Z] Invalid copilot token: missing token: 403  [ERROR] [default] [2022-06-23T06:52:51.252Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

### DISCUSSION THREAD

**Discussion by wacanam** [community/community#19251] [Copilot Conversations]
I keep on having this error. multiple restart and logout->login, disable->enable->uninstall->install but the same error.

**Comment by dhust** [comment]
The Technical Preview is over. You'll need to pay for the service now.

**Reply by akinolu52** [reply]
it's free, you start paying by August 22

**Reply by dhust** [reply]
To be more accurate, I wouldn't consider it free. It went from a Technical Preview to 60-day Trial to pay. We are now in the 60-day trial.

**Reply by akinolu52** [reply]
okay

**Reply by OlegKotok** [reply]
and how to fix it? I have the same problem.

**Comment by mashirozx** [comment]
Same, vscode

<img width="461" alt="image" src="https://user-images.githubusercontent.com/16148054/174943473-caf03bae-95bd-4586-a1b5-b3228ea91062.png">
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/16148054/174944347-5e631e8b-94bb-48ba-855e-58d81e36de05.png">

<img width="795" alt="image" src="https://user-images.githubusercontent.com/16148054/174943497-c756a3bb-0b59-41cb-83d2-78e354a413b8.png">

Seems the web auth doesn't redirect to the auth callback URL correctly, which redirect to the bill page incorrectly.

**Reply by davecheney** [reply]
@mashirozx it looks like there is a problem with your billing (I know it says free copilot, but the billing system still checks that it _would_ be possible to bill you, even though copilot is free). Please contact support and ask them to reset your billing retry count.

**Reply by mashirozx** [reply]
@davecheney Thanks, contacting

**Reply by g-hoang** [reply]
Got an exact problem, contacted Github Support and their solution works for me:

1. After verifying that the extension is up-to-date (or simply uninstalling, then reinstalling the extension), sign out of GitHub in VS Code via the bottom-left icon

2. Please confirm that you've [signed up for the trial](https://github.com/features/copilot), then [navigate here](https://github.com/settings/profile), and verify that GitHub Copilot is present under Code, planning, and automation.

3. Under [Authorized GitHub Apps](https://github.com/settings/applications), revoke access for GitHub for VS Code

4. Sign back into GitHub in VS Code, which should additionally guide you through reauthorizing the plugin via your web browser.
